x-pack/plugins/security_solution/cypress/integration/data_sources/sourcerer.spec.ts

```diff
@@ -129,103 +132,132 @@ describe('Sourcerer', () => {
+    it('Selecting new data view does not add a modified badge', () => {
+      openTimelineUsingToggle();
+      cy.get(SOURCERER.badgeModified).should(`not.exist`);
+      openSourcerer('timeline');
+      cy.get(SOURCERER.badgeModifiedOption).should(`not.exist`);
+      openDataViewSelection();
+      isKibanaDataViewOption(dataViews);
+      cy.get(SOURCERER.selectListDefaultOption).should(`contain`, siemDataViewTitle);
+      cy.get(SOURCERER.selectListOption).contains(dataViews[1]).click();
+      isDataViewSelection(dataViews[1]);
+      saveSourcerer();
+      cy.get(SOURCERER.badgeModified).should(`not.exist`);
+      openSourcerer('timeline');
+      cy.get(SOURCERER.badgeModifiedOption).should(`not.exist`);
     });
 
-    it('correctly loads SIEM data view before and after signals index exists', () => {
+    it('shows modified badge when index patterns change and removes when reset', () => {
       openTimelineUsingToggle();
       openSourcerer('timeline');
-      isDataViewSelection(siemDataViewTitle);
+      openDataViewSelection();
+      cy.get(SOURCERER.selectListOption).contains(dataViews[1]).click();
+      isDataViewSelection(dataViews[1]);
       openAdvancedSettings();
-      isSourcererSelection(`auditbeat-*`);
-      isNotSourcererSelection(`${DEFAULT_ALERTS_INDEX}-default`);
-      isSourcererOptions(
-        [...DEFAULT_INDEX_PATTERN, `${DEFAULT_ALERTS_INDEX}-default`].filter(
-          (pattern) => pattern !== 'auditbeat-*'
-        )
+      const patterns = dataViews[1].split(',');
+      deselectSourcererOptions([patterns[0]]);
+      saveSourcerer();
+      cy.get(SOURCERER.badgeModified).should(`exist`);
+      openSourcerer('timeline');
+      cy.get(SOURCERER.badgeModifiedOption).should(`exist`);
+      resetSourcerer();
+      saveSourcerer();
+      cy.get(SOURCERER.badgeModified).should(`not.exist`);
+      openSourcerer('timeline');
+      cy.get(SOURCERER.badgeModifiedOption).should(`not.exist`);
+      isDataViewSelection(siemDataViewTitle);
+    });
+  });
+  describe('Alerts checkbox', () => {
+    before(() => {
+      createTimeline(getTimeline()).then((response) =>
+        cy.wrap(response.body.data.persistTimeline.timeline.savedObjectId).as('timelineId')
       );
+      createTimeline(getTimelineModifiedSourcerer()).then((response) =>
+        cy.wrap(response.body.data.persistTimeline.timeline.savedObjectId).as('auditbeatTimelineId')
+      );
+    });
+    beforeEach(() => {
+      loginAndWaitForPage(TIMELINES_URL);
       waitForAlertsIndexToExist();
-      isSourcererOptions(DEFAULT_INDEX_PATTERN.filter((pattern) => pattern !== 'auditbeat-*'));
-      isNotSourcererOption(`${DEFAULT_ALERTS_INDEX}-default`);
     });
-
-    describe('Modified badge', () => {
-      it('Selecting new data view does not add a modified badge', () => {
-        openTimelineUsingToggle();
+    it('Modifies timeline to alerts only, and switches to different saved timeline without issue', function () {
```

Comment:
This is the new test, I reorganized a little bit while I was here. Also new are the `createTimeline` calls in the `before`